cwf/gateway/services/uesim/servicers/uesim_hssless.go

```diff
@@ -0,0 +1,133 @@
+	"golang.org/x/net/context"
+)
+
+const (
+	imsiPrefix    = "IMSI"
+	defaultUeIp   = "192.168.88.88"
+	defaultApn    = "wifi"
+	KiloBytes     = 1024
+	defaultMSISDN = "5100001234"
+)
+
+// UESimServerHssLess tracks all the UEs being simulated.
+type UESimServerHssLess struct {
+	store blobstore.BlobStorageFactory
+	cfg   *UESimConfig
+}
+
+// NewUESimServerHssLess initializes a UESimServer with an empty store map.
+// Output: a new UESimServerHssLess
+func NewUESimServerHssLess(factory blobstore.BlobStorageFactory) (*UESimServerHssLess, error) {
+	config, err := GetUESimConfig()
+	if err != nil {
+		return nil, err
+	}
+	return &UESimServerHssLess{
+		store: factory,
+		cfg:   config,
+	}, nil
+
+}
+
+func (srv *UESimServerHssLess) AddUE(ctx context.Context, ue *cwfprotos.UEConfig) (*protos.Void, error) {
+	ret := &protos.Void{}
+
+	err := validateUEData(ue)
+	if err != nil {
+		err = ConvertStorageErrorToGrpcStatus(err)
+		return ret, err
+	}
+
+	err = validateUEDataForHssLess(ue)
+	if err != nil {
+		err = ConvertStorageErrorToGrpcStatus(err)
+		return ret, err
+	}
+	addUeToStore(srv.store, ue)
+
+	return ret, nil
+}
+func (srv *UESimServerHssLess) Disconnect(ctx context.Context, id *cwfprotos.DisconnectRequest) (*cwfprotos.DisconnectResponse, error) {
+
+	ue, err := getUE(srv.store, id.GetImsi())
+	if err != nil {
+		return &cwfprotos.DisconnectResponse{}, err
+	}
+
+	req := &lte_protos.LocalEndSessionRequest{
+		Sid: makeSID(ue.GetImsi()),
+		Apn: defaultApn,
+	}
+	_, err = session_manager.EndSession(req)
+	if err != nil {
+		return &cwfprotos.DisconnectResponse{}, err
+	}
+	return &cwfprotos.DisconnectResponse{}, nil
+
+}
+func (srv *UESimServerHssLess) Authenticate(ctx context.Context, id *cwfprotos.AuthenticateRequest) (*cwfprotos.AuthenticateResponse, error) {
+
+	ue, err := getUE(srv.store, id.GetImsi())
+	if err != nil {
+		return &cwfprotos.AuthenticateResponse{SessionId: ""}, err
+	}
+
+	req := &lte_protos.LocalCreateSessionRequest{
+		CommonContext: &lte_protos.CommonSessionContext{
+			Sid:     makeSID(ue.GetImsi()),
+			UeIpv4:  defaultUeIp,
+			Apn:     defaultApn,
+			Msisdn:  ([]byte)(ue.GetMsisdn()),
+			RatType: lte_protos.RATType_TGPP_WLAN,
+		},
+		RatSpecificContext: &lte_protos.RatSpecificContext{
+			Context: &lte_protos.RatSpecificContext_WlanContext{
+				WlanContext: &lte_protos.WLANSessionContext{
+					MacAddrBinary:   ([]byte)(srv.cfg.brMac),
+					MacAddr:         srv.cfg.brMac,
+					RadiusSessionId: "sessiond1",
+				},
+			},
+		},
+	}
+	resp, err := session_manager.CreateSession(req)
+	if err == nil {
+		return &cwfprotos.AuthenticateResponse{SessionId: resp.GetSessionId()}, nil
+	}
+	return &cwfprotos.AuthenticateResponse{SessionId: ""}, err
+}
+func (srv *UESimServerHssLess) GenTraffic(ctx context.Context, req *cwfprotos.GenTrafficRequest) (*cwfprotos.GenTrafficResponse, error) {
+	glog.V(2).Infof("Reached GenTraffic...")
+	return nil, nil
+}
+
+func makeSID(imsi string) *lte_protos.SubscriberID {
```

Comment:
lets use `makeSubscriberId` 
Let's reduce any confusion with sessionId and subscriberId.

Reply:
Done